src/components/Upload.tsx

```diff
@@ -98,11 +106,13 @@ export default function Upload({ setInUploadView }: UploadProps) {
 
   return (
     <>
-      {(!ct || !rtstruct || !rtdose) && <h2>Add:</h2>}
+      {(!ctFiles.length || !rtstructFiles.length || !rtdoseFiles.length) && (
+        <h2>Add:</h2>
+      )}
       <ul style={{ listStylePosition: "inside" }}>
-        {!ct && <li>CT</li>}
-        {!rtstruct && <li>RTSTRUCT </li>}
-        {!rtdose && <li>RTDOSE </li>}
+        {!ctFiles.length && <li>CT</li>}
+        {!rtstructFiles.length && <li>RTSTRUCT </li>}
+        {!rtdoseFiles.length && <li>RTDOSE </li>}
```

Comment:
Myślę, że trochę bardziej elegantnym podejściem było by użycie w tym miejscu stanu reprezentowanego jako obiekt (można to osiągnąć za pomocą funkcji `useReducer`)

Stan wtedy mógłby wyglądać tak:

```ts
interface State {
   ctFiles: File[];
   rtStructFile: File[];
   rtDoseFile: File[];
}
```

Wtedy ten fragment mógłby wyglądać mniej więcej tak:
(zakładając, że zmienna `state` ma taką strukturę jak wyżej)
```ts

{(Object.values(state).some(collection => collection.length !== 0) && <h2>Add: </h2>}
{(Object.entries(state).filter(([key, value] => value.lenth !== 0).map(([key, value]) => <li> {key} </li>)

```

Zajmuje trochę mniej miejsca i myśle, że jest zlekka bardziej czytelne